frontend/src/concepts/modelRegistry/apiHooks/useModelRegistries.ts

```diff
@@ -1,11 +1,65 @@
 import React from 'react';
-import useFetchState, { FetchState } from '~/utilities/useFetchState';
-import { ModelRegistryKind } from '~/k8sTypes';
-import { listModelRegistries } from '~/api';
+import useFetchState, { FetchState, FetchStateCallbackPromise } from '~/utilities/useFetchState';
+import { AccessReviewResourceAttributes, ServiceKind } from '~/k8sTypes';
+import {
+  ServiceModel,
+  useAccessReview,
+  useRulesReview,
+  useServicesByNames,
+  useServices,
+} from '~/api';
+import { MODEL_REGISTRY_DEFAULT_NAMESPACE } from '~/concepts/modelRegistry/const';
 
-const useModelRegistries = (): FetchState<ModelRegistryKind[]> => {
-  const getModelRegistries = React.useCallback(() => listModelRegistries(), []);
-  return useFetchState<ModelRegistryKind[]>(getModelRegistries, []);
+const accessReviewResource: AccessReviewResourceAttributes = {
+  group: 'user.openshift.io',
+  resource: 'services',
+  verb: 'list',
+  namespace: MODEL_REGISTRY_DEFAULT_NAMESPACE,
 };
 
-export default useModelRegistries;
+const useModelRegistryServices = (): FetchState<ServiceKind[]> => {
+  const [allowList, accessReviewLoaded] = useAccessReview(accessReviewResource);
+  const [loaded, statuses] = useRulesReview(MODEL_REGISTRY_DEFAULT_NAMESPACE);
+  const serviceNames = React.useMemo(() => {
+    if (!loaded) {
+      return [];
+    }
+    return (
+      statuses?.resourceRules.find(
+        ({ resources, verbs }) =>
+          resources?.includes(ServiceModel.plural) && verbs.some((verb) => verb === 'get'),
+      )?.resourceNames || []
```

Comment:
Ok @manaswinidas I apologize, I was wrong about the logic here. After granting permission to regularuser1 to see dallas-mr on our shared cluster, I found that when I log in as regularuser1 only dallas-mr is appearing in the dropdown and modelregistry-sample disappeared. I inspected the SelfSubjectRulesReview status we get back and I see the culprit here -- It's not just one item in  `resourceRules` that includes multiple service names, each appears to get its own item there:

<img width="410" alt="Screenshot 2024-07-26 at 12 01 03 PM" src="https://github.com/user-attachments/assets/4c24f636-eae4-4f37-8c49-1cce4fa700cd">

I'm pretty sure that if you just replace the `.find(` with `.flatMap(` on line 28 here you'll get the right result.